test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/JsonOutputFormatterTests.cs

```diff
@@ -77,6 +77,33 @@ public void Constructor_UsesSerializerSettings()
         }
 
         [Fact]
+        public async Task Using_TypeNameHandlingAuto_AppliesTypeNameToRoot()
+        {
+            // Arrange
+            var person = new User() { FullName = "John", age = 35 };
+            var outputFormatterContext = GetOutputFormatterContext(person, typeof(object));
+
+            var settings = new JsonSerializerSettings
+            {
+                TypeNameHandling = TypeNameHandling.Auto
```

Comment:
In addition to @Eilon's [comments](https://github.com/aspnet/Mvc/issues/6082#issuecomment-327967724), note this test succeeds when this line is commented out. `expectedOutput` should be a `string` or the `byte`s for a string that's in the test e.g.
``` json
var expectedOutput = Encoding.UTF8.GetBytes(@"{""$type"":""Microsoft.AspNetCore.Mvc.Formatters.JsonOutputFormatterTests+User, Microsoft.AspNetCore.Mvc.Formatters.Json.Test"",""FullName"":""John"",""age"":35}";
```